src/objective-c/GRPCClient/GRPCCall.m

```diff
@@ -195,12 +183,10 @@ - (void)startNextRead {
   __weak GRPCDelegateWrapper *weakWriteable = _responseWriteable;
 
   dispatch_async(_callQueue, ^{
-    [weakSelf startReadWithHandler:^(grpc_event *event) {
-      if (!event->data.read) {
-        // No more responses from the server.
+    [weakSelf startReadWithHandler:^(NSData *data) {
+      if (data == nil) {
```

Comment:
`data == nil` is the same as `!data` (3 lines below). We still want to distinguish the two cases: there are no more responses from the server, or there was one but we don't have enough memory to hold it.